packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -320,31 +334,95 @@ function WorklistDataGrid(props) {
       // Esc (while in row selection mode and rows are selected): Clear selection
       if (onClearSelectedRows) {
         onClearSelectedRows();
       }
     } else if (onDisableSelectableRows) {
-      setAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.row-selection-mode-disabled' }));
+      rowSelection.current.focussedRow = null;
+      message = intl.formatMessage({ id: 'Terra.worklist-data-grid.row-selection-mode-disabled' });
       onDisableSelectableRows();
     }
+    setAriaLiveMessage(message);
+  };
+
+  const selectMultipleRows = (newEndRowIndex, selectionData) => {
+    if (!onRowSelect || !selectionData) {
+      return;
+    }
+
+    if (!hasSelectableRows) {
+      onEnableRowSelection();
+    }
+
+    // Rows in range to remain selected
+    const selectionStartRowIndex = Math.min(selectionData.anchorRow, newEndRowIndex);
+    const selectionEndRowIndex = Math.max(selectionData.anchorRow, newEndRowIndex);
+
+    // We are subtracting 1 to accommodate for the column header in the grid.
+    const rowsToSelect = rows.slice(selectionStartRowIndex - 1, selectionEndRowIndex);
+    const rowIdsToSelect = rowsToSelect.map(r => ({ rowId: r.id, isSelected: true }));
+
+    // Determine if there are rows that are no longer in range that need to be unselected.
+    let rowIdsToUnselect = [];
+    if (selectionData.anchorRow > selectionData.previousSelectionEndRow) {
+      // The range extends upward from the anchor row
+      if (newEndRowIndex > selectionData.previousSelectionEndRow) {
+        // The range was moved down towards the anchor so rows that no longer qualify for the range need to be unselected.
+        const rowsToUnselect = rows.slice(selectionData.previousSelectionEndRow - 1, newEndRowIndex);
+        rowIdsToUnselect = rowsToUnselect.map(r => ({ rowId: r.id, isSelected: false }));
+      }
+    } else if (selectionData.anchorRow < selectionData.previousSelectionEndRow) {
+      // The range extends downward from the anchor row
+      if (newEndRowIndex < selectionData.previousSelectionEndRow) {
+        // The range was moved up towards the anchor so rows that no longer qualify for the range need to be unselected       // New endRangeIndex becomes ordered End
+        const rowsToUnselect = rows.slice(newEndRowIndex - 1, selectionData.previousSelectionEndRow);
+        rowIdsToUnselect = rowsToUnselect.map(r => ({ rowId: r.id, isSelected: false }));
+      }
+    }
+
+    let ariaMessage = intl.formatMessage({
+      id: hasSelectableRows
+        ? 'Terra.worklist-data-grid.row-selection-multiple-rows-selected'
+        : 'Terra.worklist-data-grid.row-selection-mode-enabled',
+    });
+
+    ariaMessage += intl.formatMessage({
+      id: rowIdsToSelect.length === 1
+        ? 'Terra.worklist-data-grid.row-selection-template'
+        : 'Terra.worklist-data-grid.row-selection-selected-rows-range',
+    }, { row: selectionData.anchorRow, endRow: newEndRowIndex });
```

Comment:
How are you separating these two sentences?  There is no punctuation between them.

Reply:
Added period between two sentences.